src/main/java/org/folio/circulation/services/PubSubPublishingService.java

```diff
@@ -66,4 +69,42 @@ public PubSubPublishingService(RoutingContext routingContext) {
 
     return publishResult;
   }
+
+  /**
+   * Method that publishes event to PubSub and includes response body in the exception message in
+   * case of any response status other than 204.
+   *
+   * @param eventMessage Event to be published
+   * @param params Okapi connection parameters
+   * @return
+   */
+  private CompletableFuture<Boolean> sendEventMessage(Event eventMessage,
+    OkapiConnectionParams params) {
+
+    PubsubClient client = new PubsubClient(params.getOkapiUrl(), params.getTenantId(),
+      params.getToken());
+    CompletableFuture<Boolean> result = new CompletableFuture<>();
+
+    try {
+      client.postPubsubPublish(eventMessage, (ar) -> {
+        if (ar.statusCode() == HttpStatus.HTTP_NO_CONTENT.toInt()) {
+          result.complete(true);
+        } else {
+          ar.bodyHandler(body -> {
+            EventSendingException exception = new EventSendingException(String.format(
+            "Error during publishing Event Message in PubSub. " +
+              "Status code: %s . Status message: %s . Response body: %s",
+            ar.statusCode(), ar.statusMessage(), body.toString()));
+            logger.error(exception);
+            result.completeExceptionally(exception);
+          });
+        }
+      });
+      return result;
+    } catch (Exception var5) {
+      logger.error("Error during sending event message to PubSub", var5);
```

Comment:
```suggestion
    } catch (Exception ex) {
      logger.error("Error during sending event message to PubSub, event {} ", ex, eventMessage.getPayload());
```

Reply:
Done, but it dropped the test coverage under 80%, I'll try to cover this piece with tests.